### DISCUSSION TITLE
Add VOID and UNCOLLECTIBLE statuses for invoices

### DISCUSSION THREAD

**Discussion by sebastiancretu** [crater-invoice-inc/crater#559] [Ideas]
Maybe we should add 2 more invoice statuses: VOID and UNCOLLECTIBLE.

**Voided invoices** 
Voiding an invoice should be conceptually similar to deleting (canceling) it or a soft delete. However, voiding an invoice maintains a paper trail, which will allow us to look up the invoice by number. Voided invoices would be treated as zero-value for reporting purposes. 
We should be able to void invoices only when they’re in either the **SENT** or **UNCOLLECTIBLE** status.

**Uncollectible invoices** 
Sometimes we have customers who can’t pay their overdue bills. For example, assume that you supply 1000 EUR worth of services to your customer, but they’ve since declared bankruptcy, and (even after liquidating assets) won’t be able to pay the outstanding invoice balance.
As a result, you decide to write off the invoice as unlikely to be paid. In this scenario, you can update the status of the invoice to be uncollectible. This allows you to track the amount owed for reporting purposes as part of your bad-debt accounting process

We could have the following workflow:

![crater](https://user-images.githubusercontent.com/2368005/133922603-64fd7a7a-ccf5-4169-8201-303b32550545.png)

**Comment by patricklucas** [comment]
I think this feature makes a lot of sense and shouldn't be too difficult to implement.
